README.md

```diff
@@ -1,2 +1,113 @@
 # activerecord-dynamic_timeout
-Dynamic Timeouts within ActiveRecord
+Dynamic query timeouts within ActiveRecord!
+
+## Installation
+
+Add this line to your application's Gemfile:
+```ruby
+gem 'activerecord-dynamic_timeout', require: 'active_record/dynamic_timeout/railtie'
+```
+
+### Non-Rails Applications
+If you are using this gem in a non-rails application, you will need run the initializer manually.
+```ruby
+require "active_record/dynamic_timeout"
+ActiveRecord::DynamicTimeout::Initializer.initialize!
+```
+
+## Usage
+    
+```ruby
+class MyModel < ActiveRecord::Base
+  # Model code...
+end
+
+MyModel.with_timeout(5) do
+  MyModel.find(1)
+end
+```
+
+## Supported Adapters
+* mysql2
+* trilogy - ActiveRecord versions 7.1+
+* postgresql
+* sqlite3
+
+See [OtherAdapters](#other-adapters) on how to add support for other adapters.
+
+### Mysql2
+Timeouts are set using the client read_timeout and write_timeout attributes on the client. At the moment this is a bit of a hack as the mysql2 gem doesn't provide
+a clean way to set these attributes (via a public method).
```

Comment:
So I dug into the `MAX_EXECUTION_TIME` system variable more... It turns out this 100% can return an error. It just turns out that `SELECT SLEEP(2)` with an execution time of 2 seconds will just return `1` and not raise an error. However if we attempt to query an actual table and it takes longer than the execution time, this will work as expected.